scripts/install.js

```diff
@@ -45,13 +47,9 @@ function applyProxy(options, cb) {
     var proxyUrl;
 
     if (!er) {
-      ['https-proxy', 'proxy', 'http-proxy'].some(function(setting) {
-        var npmProxyUrl = conf.get(setting);
-        if (npmProxyUrl) {
-          proxyUrl = npmProxyUrl;
-          return true;
-        }
-      });
+      proxyUrl = ['https-proxy', 'proxy', 'http-proxy'].filter(function(setting) {
```

Comment:
Agree that is much better. :+1: 

Fixed by 63331e0.

For better approach in the scenario, where the given array's size is larger, see http://stackoverflow.com/q/28476629/. Lets see what kind of responses we get there! :)